Shared/sdk/SharedUtil.File.hpp

```diff
@@ -1052,9 +1068,63 @@ int SharedUtil::File::Delete(const char* szFilename)
 
 int SharedUtil::File::Rename(const char* szOldFilename, const char* szNewFilename)
 {
-#ifdef WIN32
+#ifdef _WIN32
     return _wrename(FromUTF8(szOldFilename), FromUTF8(szNewFilename));
 #else
     return rename(szOldFilename, szNewFilename);
 #endif
 }
+
+std::vector<std::string> SharedUtil::ListDir(const char* szPath) noexcept
+{
+    std::vector<std::string> entries;
+#if __cplusplus >= 201703L
+    namespace fs = std::filesystem;
+    if (!DirectoryExists(szPath))
+        return {};
+
+    try
+    {
+        for (const auto& entry : fs::directory_iterator(szPath))
+        {
+            if (entry.is_regular_file() || entry.is_directory())
+                entries.push_back(entry.path().filename().string());
+        }
+    }
+    catch (...) {} // catch all possible errors and ignore them
+#else
+    #ifdef _WIN32
+    std::string search_path = szPath;
+    if (search_path.empty())
+        return {};
+
+    if (search_path.back() != '/')
+        search_path += "/*";
+
+    WIN32_FIND_DATA fd;
+    HANDLE          hFind = ::FindFirstFile(search_path.c_str(), &fd);
+    if (hFind == INVALID_HANDLE_VALUE)
+        return {};
+
+    do
+    {
+        entries.push_back(fd.cFileName);
+    } while (::FindNextFile(hFind, &fd));
+    ::FindClose(hFind);
+
+    #else
+    DIR*           dir;
+    struct dirent* ent;
+    if (!(dir = opendir(szPath)))
+        return {};
+
+    while ((ent = readdir(dir)))
+    {
+        entries.push_back(ent->d_name);
+    }
+    closedir(dir);
+    #endif
+#endif
+
+    return entries;
+}
```

Comment:
```suggestion

#if __cplusplus >= 201703L
std::vector<std::string> SharedUtil::ListDir(const char* szPath) noexcept
{
    std::vector<std::string> entries;
    namespace fs = std::filesystem;
    if (!DirectoryExists(szPath))
        return {};

    try
    {
        for (const auto& entry : fs::directory_iterator(szPath))
        {
            if (entry.is_regular_file() || entry.is_directory())
                entries.push_back(entry.path().filename().string());
        }
    }
    catch (...) {} // catch all possible errors and ignore them

    return entries;
}
#endif
```
I think there is no reason to add dead code to the project.

Reply:
Some mta projects are still limited to c++14. Better safe than sorry

Reply:
But this code is unused and should be removed when we get C++ 17 there.
The codebase becomes more complex due to dead code. I would recommend keeping only the necessary code that can be tested.

Reply:
After the most recent force push, have you made these changes back?

Reply:
> After the most recent force push, have you made these changes back?

Nope. The code is used by `multiplayer_sa`. Cannot remove it